k8s/src/main/scala/io/buoyant/k8s/resources.scala

```diff
@@ -0,0 +1,308 @@
+package io.buoyant.k8s
+
+import com.twitter.concurrent.AsyncStream
+import com.twitter.finagle.http.Response
+import com.twitter.finagle.{Filter, http}
+import com.twitter.finagle.param.HighResTimer
+import com.twitter.finagle.service.{Backoff, RetryBudget, RetryFilter, RetryPolicy}
+import com.twitter.finagle.stats.{DefaultStatsReceiver, StatsReceiver}
+import com.twitter.io.{Buf, Reader}
+import com.twitter.util.{Closable, _}
+import com.twitter.util.TimeConversions._
+import java.util.concurrent.atomic.AtomicReference
+
+/**
+ * Basic functionality shared by all k8s Resource implementations: they require an HTTP client, and can be closed.
+ */
+trait Resource extends Closable {
```

Comment:
maybe name this file `Resource.scala` to match this trait?  I know there are a bunch of resources in this file, but it might be good to have the file name match at least one of the things defined in it.